src/main/java/com/zendesk/maxwell/monitoring/MaxwellHTTPServer.java

```diff
@@ -0,0 +1,143 @@
+		String reportingType = config.metricsReportingType;
+		if (reportingType != null && reportingType.contains(reportingTypeHttp)) {
+			return new MaxwellMetrics.Registries(config.metricRegistry, config.healthCheckRegistry);
+		} else {
+			return null;
+		}
+	}
+
+	private static MaxwellDiagnosticContext getDiagnosticContext(MaxwellContext context) {
+		MaxwellDiagnosticContext.Config diagnosticConfig = context.getConfig().diagnosticConfig;
+		if (diagnosticConfig.enable) {
+			return new MaxwellDiagnosticContext(diagnosticConfig, context.getDiagnostics());
+		} else {
+			return null;
+		}
+	}
+}
+
+class MaxwellHTTPServerWorker implements StoppableTask, Runnable {
+
+	private int port;
+	private final String pathPrefix;
+	private final MaxwellMetrics.Registries metricsRegistries;
+	private final MaxwellDiagnosticContext diagnosticContext;
+	private Server server;
+
+	public MaxwellHTTPServerWorker(int port, String pathPrefix, MaxwellMetrics.Registries metricsRegistries,
+																 MaxwellDiagnosticContext diagnosticContext) {
+		this.port = port;
+		this.pathPrefix = pathPrefix;
+		this.metricsRegistries = metricsRegistries;
+		this.diagnosticContext = diagnosticContext;
+	}
+
+	public void startServer() throws Exception {
+		this.server = new Server(this.port);
+		ServletContextHandler handler = new ServletContextHandler(this.server, "/");
+
+		if (metricsRegistries != null) {
+			// TODO: there is a way to wire these up automagically via the AdminServlet, but it escapes me right now
+			handler.addServlet(new ServletHolder(new MetricsServlet(metricsRegistries.metricRegistry)), genFullPath(pathPrefix, "/metrics"));
+			handler.addServlet(new ServletHolder(new HealthCheckServlet(metricsRegistries.healthCheckRegistry)), genFullPath(pathPrefix, "/healthcheck"));
+			handler.addServlet(new ServletHolder(new PingServlet()), genFullPath(pathPrefix, "/ping"));
+		}
+
+		if (diagnosticContext != null) {
+			handler.addServlet(new ServletHolder(new DiagnosticHealthCheck(diagnosticContext)),
+					genFullPath(pathPrefix, diagnosticContext.config.path));
+		}
+
+		this.server.start();
+		this.server.join();
+	}
+
+	@Override
+	public void run() {
+		try {
+			startServer();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override
+	public void requestStop() {
+		try {
+			this.server.stop();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Override
+	public void awaitStop(Long timeout) throws TimeoutException {
+	}
+
+	private String genFullPath(String pathPrefix, String path) {
+		String prefix = genPath(pathPrefix);
+		if (prefix != null) {
+			return prefix + genPath(path);
+		} else {
+			return genPath(path);
+		}
+	}
+
+	private String genPath(String path) {
+		if (StringUtils.isNotBlank(path)) {
+			if (!path.startsWith("/")) {
+				return "/" + path;
+			}
+			return path;
+		} else {
+			return null;
+		}
+	}
```

Comment:
These two methods allow for a lot of leeway (nulls, empty, slashes in various places). Could we instead normalize the input configuration, so that:

 - `pathPrefix` is always a non-null string which starts and ends with a slash, ie.. `"/"` or `"/some/path/"`
 - `path` never starts with a slash

That way string concat will give you nice paths without needing to think about the possible variations.

I haven't checked, but it's also possible we could put the prefix in the call to `new ServletContextHandler`, and then we just make sure that it handles prefixes sanely.